app/routes/surveys/components/Question.js

```diff
@@ -0,0 +1,237 @@
+        </Link>
+      </Dropdown.ListItem>
+      <Dropdown.ListItem>
+        <Link onClick={() => setQuestionType(2)}>
+          Sjekkboks
+          <Icon name="checkbox" size={24} />
+        </Link>
+      </Dropdown.ListItem>
+      <Dropdown.ListItem>
+        <Link onClick={() => setQuestionType(3)}>
+          Fritekst
+          <Icon name="more" size={24} />
+        </Link>
+      </Dropdown.ListItem>
+    </Dropdown.List>
+  );
+}
+
+class Question extends Component<Props, State> {
+  state = {
+    shouldUpdate: false,
+    options: [{ nr: 0, optionText: '' }],
+    pickerOpen: false
+  };
+
+  componentDidMount() {
+    const options = this.props.question.options
+      .map((option, i) => ({ ...option, nr: i }))
+      .concat({
+        nr: this.props.question.options.length,
+        optionText: ''
+      });
+    this.setState({ options });
+  }
+
+  componentDidUpdate() {
+    if (this.state.shouldUpdate) {
+      this.props.updateQuestion({
+        ...this.props.question,
+        options: this.state.options
+      });
+      this.setState({ shouldUpdate: false });
+    }
+  }
+
+  updateOptions = (option: Object) => {
+    const options = this.state.options.slice();
+
+    const oldOption = options.find(o => o.nr === option.nr);
+    const changedOptionIndex = options.indexOf(oldOption || {});
+
+    if (changedOptionIndex !== -1) {
+      // This inner "if" adds another empty option below the current one if
+      // the current option is the last one, which is how several options
+      // functionality is implemented.
+      if (
+        option.nr === options.length - 1 &&
+        (oldOption || {}).optionText === ''
+      ) {
+        options.push({
+          optionText: '',
+          nr: this.state.options.length
+        });
+      }
+      // Afterwards, outside that inner if, the current option is then updated.
+      options[changedOptionIndex] = option;
+    } else {
+      options.push(option);
+    }
+
+    this.setState({ options, shouldUpdate: true });
+  };
+
+  toggleDropdown = () => {
+    this.setState(prevState => ({
+      pickerOpen: !prevState.pickerOpen
+    }));
+  };
+
+  setQuestionType = (questionType: number) => {
+    this.props.updateQuestion({ ...this.props.question, questionType });
+  };
+
+  render() {
+    const { updateQuestion, nr, question, deleteQuestion } = this.props;
+    return (
+      <div className={styles.question}>
+        <div className={styles.left}>
+          <div className={styles.questionTop}>
+            <TextInput
+              value={question.questionText}
+              className={styles.questionTitle}
+              placeholder="Spørsmål"
+              onInput={e =>
+                updateQuestion({
+                  ...this.props.question,
+                  questionText: e.target.value,
+                  nr
+                })
+              }
+            />
+          </div>
+          {this.props.question.questionType === 3 ? (
```

Comment:
It'd be nice to use something enum like for these question type constants, i.e. something like

```js
const QUESTION_TYPE = {
  TEXT: 1,
  EXISTENTIAL: 2,
  BLABLA: 3
};
```

Reply:
Yeah I figured we needed some more obvious representation, I just wasn't sure if the current standard we used was an object like you suggest or just changing the whole variable to a string or something else. So I held off on commiting to either one, but if you think objects like this should be the way to go then I'll gladly use that.